opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -288,6 +288,20 @@ def test_start_span(self):
         span.start()
         self.assertEqual(start_time, span.start_time)
 
+        # default status
+        self.assertTrue(span.status.is_ok)
+        self.assertEqual(
+            span.status.canonical_code, trace_api.status.StatusCanonicalCode.OK
+        )
+        self.assertIs(span.status.description, None)
+
+        # status
+        new_status = trace_api.status.Status(
+            trace_api.status.StatusCanonicalCode.CANCELLED, "Test description"
+        )
+        span.set_status(new_status)
+        self.assertEqual(span.status, new_status)
```

Comment:
You don't override `__hash__` and `__eq__` in `Status`, so you are effectively checking `assertIs` here. I think for status it would actually make sense to override these (also `__repr__`).

Reply:
I will update the test to check for all properties in the object instead, we can follow this in a different PR and discuss if we want to do this for other objects in the code